src/config/app-config-endpoints.html

```diff
@@ -6,6 +6,39 @@
     domain: '.domain.com'
   };
 
+  function polyfillLocationOrigin() {
+    if (window.location.origin) { return; }
+
+    window.location.origin = window.location.protocol + "//" + window.location.hostname + (window.location.port ? ":" + window.location.port : "");
+  }
+
+  function resolveApiBaseUrl() {
```

Comment:
So based on the cases you sent me:
localhost:8080 => d-safeco-api.salesnext.com
q-safeco.localhost:8080 => q-safeco-api.salesnext.com
q-safeco.salesnext.com => q-safeco-api.salesnext.com

We can build out a pattern of: [subdomain.]domain => subdomain-api.domain, and we're always "rebuilding" apiBaseUrl, so with that in mind:

```
const config = AppEndpoints;
let [, protocol, , subdomain, domain] = /^([a-z]*:\/\/)?(([a-zA-Z0-9\-]*)[.])?([.]*[^:]*)([:][a-zA-Z0-9]*)?$/.exec(window.location.origin);

// No subdomain, use the supplied one
if (!subdomain) subdomain = config.subdomain;
// Otherwise add -api if it's missing
else if (!subdomain.endsWith('-api')) subdomain.concat('-api');

// If we're using localhost then we need to swap the protocol and domain
if (domain.endsWith('localhost')) {
  domain = config.domain;
  delete protocol;
}

// If there's no protocol, use the default
if (!protocol) protocol = config.protocol;

config.baseUrl = `${protocol}${subdomain}.${domain}${port}`;
```